src/shared/sam/debugger/awsSamDebugConfigurationValidator.ts

```diff
@@ -12,96 +13,120 @@ import {
-
-    public validateSamDebugConfiguration(debugConfiguration: AwsSamDebuggerConfiguration): ValidationResult {
-        const generalValidationResult = this.generalDebugConfigValidation(debugConfiguration)
-        if (!generalValidationResult.isValid) {
-            return generalValidationResult
+    public constructor(
+        private readonly cftRegistry = CloudFormationTemplateRegistry.getRegistry(),
+        private readonly workspaceFolder: vscode.WorkspaceFolder | undefined
+    ) {}
+
+    /**
+     * Validates debug configuration properties.
+     */
+    public validate(config: AwsSamDebuggerConfiguration): boolean {
+        let rv: { isValid: boolean; message?: string } = { isValid: false, message: undefined }
+        if (!config.request) {
+            rv.message = localize(
+                'AWS.sam.debugger.missingField',
+                'Missing required field "{0}" in debug config',
+                'request'
+            )
+        } else if (!AWS_SAM_DEBUG_REQUEST_TYPES.includes(config.request)) {
+            rv.message = localize(
+                'AWS.sam.debugger.invalidRequest',
+                'Debug Configuration has an unsupported request type. Supported types: {0}',
+                AWS_SAM_DEBUG_REQUEST_TYPES.join(', ')
+            )
+        } else if (!AWS_SAM_DEBUG_TARGET_TYPES.includes(config.invokeTarget.target)) {
+            rv.message = localize(
+                'AWS.sam.debugger.invalidTarget',
+                'Debug Configuration has an unsupported target type. Supported types: {0}',
+                AWS_SAM_DEBUG_TARGET_TYPES.join(', ')
+            )
+        } else if (config.invokeTarget.target === TEMPLATE_TARGET_TYPE) {
+            let cfnTemplate
+            if (config.invokeTarget.samTemplatePath) {
+                const fullpath = tryGetAbsolutePath(this.workspaceFolder, config.invokeTarget.samTemplatePath)
+                // Normalize to absolute path for use in the runner.
+                config.invokeTarget.samTemplatePath = fullpath
+                cfnTemplate = this.cftRegistry.getRegisteredTemplate(fullpath)?.template
+            }
+            rv = this.validateTemplateConfig(config, config.invokeTarget.samTemplatePath, cfnTemplate)
+        } else if (config.invokeTarget.target === CODE_TARGET_TYPE) {
+            rv = this.validateCodeConfig(config)
         }
 
-        if (isTemplateTargetProperties(debugConfiguration.invokeTarget)) {
-            return this.templateDebugConfigValidation(debugConfiguration, this.cftRegistry)
-        } else if (isCodeTargetProperties(debugConfiguration.invokeTarget)) {
-            return this.codeDebugConfigValidation(debugConfiguration)
+        if (!rv.isValid) {
+            getLogger().error(`invalid debug-config: ${rv.message}`)
+            vscode.window.showErrorMessage(rv.message ?? 'invalid debug-config')
+        } else if (rv.message) {
+            vscode.window.showInformationMessage(rv.message)
```

Comment:
It shouldn't be a concern of this class to show a message to the user (vscode.window.showInformationMessage).

Reply:
this PR is a cleanup step. can we do the refactoring later?

Reply:
This PR introduced the change -- previous code didn't perform messaging at the validation level.

Reply:
This PR is merging existing logic only. The code here was moved from the old validation location.

Reply:
The code was moved from the vscode.DebugConfigurationProvider implementation, where the messaging responsibility belongs. This seems like a partial clean up step, because deferring this work affects the signature (validate return bool instead of validate return status), which will require further adjustments to methods that call this function.

Are you sure you'd like to defer this adjustment?

Reply:
moved `vscode.window.show*` calls out of `awsSamDebugConfigurationValidator.ts`